### PR TITLE
Add command line arguments to compiler

### PR DESCRIPTION
**Pull request by mayanksharma3**
### Clubhouse Ticket
[ch109](https://app.clubhouse.io/manim-dsl/story/109/add-command-line-arguments-to-compiler)

### Description

Added CLI to the compiler with help messages etc . 

Documentation to come

### Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue) :bug
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

### How Has This Been Tested?

None needed.

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by arj119** [comment]
Converting to draft till #14 and #13 have been merged

File: src/main/kotlin/com/manimdsl/Compiler.kt [hunk 1]
```diff
@@ -30,27 +33,60 @@ private fun compile(filename: String) {
...
     do {
          state = executor.executeNextStatement()
     } while (!state.first)
 
     val writer = ManimProjectWriter(ManimWriter(state.second).build())
 
-    writer.createPythonFile("test.py")
-    writer.generateAnimation("test.py")
+    if(output !== null) println("Writing file to $output")
+    val outputFile = writer.createPythonFile(output)
```
Comment (rbanerjee20) [inline src/main/kotlin/com/manimdsl/Compiler.kt:43]: What happens here if `output` is null?

Comment (mayanksharma3) [reply src/main/kotlin/com/manimdsl/Compiler.kt:43]: It creates a temp file as they didn't ask for the python file out

**Review by arj119 — DISMISSED** [review]
LGTM 👍

**Review by arj119 — APPROVED** [review]
LGTM pt 2 👍

**Review by arj119 — APPROVED** [review]
LGTM pt 3 👍

**Pull request merged**